logger/example_test.go

```diff
@@ -0,0 +1,93 @@
+package logger
+
+import (
+	"fmt"
+)
+
+func ExampleMaskJSON() {
+	maskData := []MaskData{
+		{
+			JSONKeys:         []string{"first"},
+			FirstCharsToShow: 3,
+			LastCharsToShow:  6,
+			KeepSameLength:   true,
+		},
+		{
+			JSONKeys:         []string{"second", "second"},
+			FirstCharsToShow: 2,
+			LastCharsToShow:  3,
+			CharsToIgnore:    []rune{'@'},
+			RestrictionType:  MaskRestrictionTypeEmail,
+			KeepSameLength:   true,
+		},
+		{
+			JSONKeys:         []string{"second", "third", "first"},
+			FirstCharsToShow: 3,
+			LastCharsToShow:  1,
+			KeepSameLength:   true,
+		},
+		{
+			JSONKeys:         []string{"second", "fourth", "[]", "email", "value"},
+			FirstCharsToShow: 1,
+			LastCharsToShow:  3,
+			CharsToIgnore:    []rune{'@'},
+			KeepSameLength:   true,
+		},
+	}
+
+	input := `
+{
+  "first": "first value",
+  "second": {
+    "first": "1st of second",
+    "second": "not-an-email.com",
+    "third": {
+      "first": "1st of second third",
+      "second": "2nd of second third",
+      "third": "3rd of second third"
+    },
+    "fourth": [
+      { "email": { "value": "first@email.com" } },
+      { "email": { "value": "second@email.com" } }
+    ]
+  }
+}	
+`
+
+	output := MaskJSON(input, "*", maskData)
+	_, _ = fmt.Println(output)
```

Comment:
Revive mentioned to handle the `err` of `fmt.Println()`. We can't just use `fmt.Println()` directly unless we add `revive:disable` before and `revive:enable` after this line,